gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/activity/impl/GenerateWorkUnitsImpl.java

```diff
@@ -75,7 +75,7 @@ public int generateWorkUnits(Properties jobProps, EventSubmitterContext eventSub
       JobStateUtils.writeWorkUnits(workUnits, workDirRoot, jobState, fs);
       JobStateUtils.writeJobState(jobState, workDirRoot, fs);
 
-      return workUnits.size();
+      return jobState.getTaskCount();
```

Comment:
since I got it wrong the first time, this likely deserves a comment.

also, I may not be logging the count anywhere, but it could be a meaningful thing to log both the MWUs and individual WUs count.  (in fact that might do double duty-duty in lieu of that comment I just suggested)